src/ecdsa/keygen.rs

```diff
@@ -809,7 +835,11 @@ impl State<KeyGeneratorTraits> for Phase3 {
             log::error!("Phase3 returns errors {:?}", errors);
             return Transition::FinalState(Err(ErrorState::new(errors)));
         }
-        let mut dk = dk.unwrap();
+        let mut dk = dk.expect("invalid paillier decryption key");
+
+        let public_key = from_secp256k1_pk(public_key.expect("cant compute public key"))
```

Comment:
With comment in line 822, the inner `expect` can be dropped

Reply:
That is true. This code is changed in another branch, because dk must be used for verification. 
 It makes sense to check final version later.